DA-1070: Add gsi tutorial for jina

VirajAgarwal-couchbase opened the pull request.

VirajAgarwal1 committed:
update: made dirs for fts and gsi

VirajAgarwal1 committed:
update: forntmatter of gsi and fts tutorial

VirajAgarwal1 committed:
update: fts tutorial link and execution results

VirajAgarwal1 committed:
added: env sample file and gsi tutorial

VirajAgarwal1 committed:
update: gsi tutorial

- added perf comparision before and after gsi and cache
- fixed sections order and organization
- score -> distance

VirajAgarwal1 committed:
update: GSI tutorial

- simplified performance testing
- added explicit composite index sub-heading

prajwal-pai77 commented on jinaai/gsi/RAG_with_Couchbase_and_Jina_AI.ipynb:
```diff
@@ -0,0 +1,1538 @@
+      "Question: What are the latest developments in football transfers?\n",
+      "Response (completed in 2.58 seconds): ### Answer:\n",
+      "\n",
+      "I'm sorry, but based on the context provided, I cannot offer information on the latest developments in football transfers. If you have any other questions, feel free to ask!\n",
```
Please use a better prompt?

prajwal-pai77 commented on jinaai/gsi/RAG_with_Couchbase_and_Jina_AI.ipynb:
```diff
@@ -0,0 +1,1538 @@
+      "Second execution (cache hit - should be faster):\n",
+      "\n",
+      "[Cache Test - Second Run] Testing vector search performance\n",
+      "[Cache Test - Second Run] Query: 'What are the latest football transfer developments?'\n",
```
Isn't this supposed to be faster?

VirajAgarwal-couchbase replied:
Thank you for this catch! I forgot to add the actual cache. Have fixed it now, and results are as expected.

prajwal-pai77 commented on jinaai/gsi/RAG_with_Couchbase_and_Jina_AI.ipynb:
```diff
@@ -0,0 +1,1538 @@
+    "   - Waits 2 seconds for collection to be ready\n",
+    "\n",
+    "Additional Tasks:\n",
+    "- Creates primary index on collection for query performance\n",
```
Please remove this

prajwal-pai77 reviewed (changes requested):
Couple of small changes & should be good to go

VirajAgarwal1 committed:
update: based on Prajwal's comments

- Added cache setup
- removed comment fo creating primary index
- Revised user queries and responses for better rag performance
